### PR TITLE
Add API documentation for the SAP Cloud SDK for Java

### PR DESCRIPTION
**Pull request by KevinHudemann**
## What Has Changed?

This PR adds a new link to the API documentation in the sidebar of the SAP Cloud SDK for Java.
It only applies to the version v5 and includes a first version of the API docs.

### CODE REVIEW

File: docs-java_versioned_docs/version-v5/features/connectivity/destinations.mdx [hunk 1]
```diff
@@ -446,7 +446,7 @@ DestinationOptions options = DestinationOptions.builder().build();
 
 ##### Retrieval Strategy Options
 
-For a provider/subscriber setup, a [retrieval strategy](https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/cloudplatform/connectivity/ScpCfDestinationRetrievalStrategy.html) must be chosen according to your particular use case from:
+For a provider/subscriber setup, a [retrieval strategy](pathname:///java-api/v5/com/sap/cloud/sdk/cloudplatform/connectivity/ScpCfDestinationRetrievalStrategy.html) must be chosen according to your particular use case from:
```
Comment (MatKuhr) [inline docs-java_versioned_docs/version-v5/features/connectivity/destinations.mdx:449]: 😱:partying_face: :heart_eyes:

File: static/java-api/v5/index.html [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    <li>Abstractions for SAP Business Technology Platform features</li>
+    <li>Virtual Data Model for the S4/HANA APIs</li>
+    <li>Connectivity to S4/HANA (both cloud and on-premise) and other systems</li>
+    <li>Tenant and principal (user) awareness</li>
+    <li>Resilient execution of requests</li>
+</ul>
+The API is structured as follows:
+<ul>
+    <li><a href="/cloudplatform/package-summary.html">cloudplatform</a> contains platform specific features such as access to services, connectivity and resilience</li>
+    <li><a href="/s4hana/package-summary.html">s4hana</a> contains VDM representations of the S/4HANA APIs</li>
```
Comment (CharlesDuboisSAP) [inline static/java-api/v5/index.html:71]: Those links don't work.

File: static/java-api/v5/index.html [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+<main role="main">
+<div class="header">
+<h1 class="title">SAP Cloud SDK - Parent POM 0.0.2 API</h1>
+</div>
+<div class="block">The SAP Cloud SDK for Java comprises a toolkit to develop cloud native applications on the SAP Business Technology Platform. In particular it makes it easy to connect to S4/HANA systems both in the cloud and on-premise but it also offers abstractions for the SAP Business Technology Platform.
+
+<h3>Key Features:</h3>
+<ul>
+    <li>Abstractions for SAP Business Technology Platform features</li>
+    <li>Virtual Data Model for the S4/HANA APIs</li>
```
Comment (CharlesDuboisSAP) [inline static/java-api/v5/index.html:63]: A lot of this home page is outdated (ex: We don't ship the VDM in v5). I will create a BLI for after v5 is released.

**Review by CharlesDuboisSAP — COMMENTED** [review]
I will create a BLI for updating the Javadoc homepage, but if it's easy to remove/update the broken links then do this before I approve

File: static/java-api/v5/index.html (hunk 3 above)
Comment (MatKuhr) [reply static/java-api/v5/index.html:63]: @CharlesDuboisSAP please note the TODO here: https://github.com/SAP/cloud-sdk-java-backlog/issues/281

File: static/java-api/v5/index.html (hunk 2 above)
Comment (CharlesDuboisSAP) [reply static/java-api/v5/index.html:71]: I will approve anyway then based on Matthias' comment

**Pull request merged**